lock/compound.go

```diff
@@ -0,0 +1,95 @@
+package lock
+
+// NewCompund creates a multi-level locking service. In order to acquire a lock
+// it uses first level 1 and then level 2. If a lock cannot be acquired on level
+// 1, level 2 will not be tried. This is useful to have a local lock to synchronize
+// multiple goroutines as well as a distributed lock to synchronize between
+// processes.
+func NewCompound(level1, level2 Service) Service {
```

Comment:
Do we have an actual use for this? If not, I will vote to do not add it by now, to not increase the amount of code to maintain.